dashboard/app/models/section.rb

```diff
@@ -94,9 +94,23 @@ def name_safe_students
         student.name.split('').each do |letter|
           leaf = leaf[letter.to_sym]
         end
+
         # we then traverse up the trie until we encounter the
-        # "rightmost" letter in the student's name which is not unique
-        leaf = leaf.parent while leaf.parent && leaf.parent.children.count == 1
+        # "rightmost" letter in the student's name which is not unique.
+        # "Not unique" means that either the letter has siblings
+        # (implying other names that share characters with our own) or
+        # its parent is a terminal node (implying a name that's a strict
+        # subset of our own)
+        leaf = leaf.parent until leaf.parent.nil? ||
+            leaf.parent.children.count > 1 ||
+            leaf.parent.terminal?
+
+        # If our "rightmost" character is the empty string, add an
+        # additional letter for visibility if we can. Note that by this
+        # stage we are guaranteed to have no more than one child, so we
+        # can condifently pick the first.
+        leaf = leaf.children[0] if leaf.children.count > 0 && leaf.letter == :" "
```

Comment:
- i prefer `.first` to `[0]`, don't know if we have an enforced/recommended style one way or another though.
- you can use `.any?` instead of `.count > 0`
- Is `:" "` the empty string, or a 'space' character? The comment refers to 'the empty string', not sure if this is what it's referring to however.